packages/node/src/typescript.ts

```diff
@@ -189,37 +189,54 @@ export function register(opts: Options = {}): Register {
     }
   }
 
-  function getBuild(configFileName = ''): Build {
-    let build = configFileToBuildMap.get(configFileName);
-    if (build) return build;
-
-    const config = readConfig(configFileName);
-
-    /**
-     * Create the basic required function using transpile mode.
-     */
-    const getOutput = function (code: string, fileName: string): SourceOutput {
-      const result = ts.transpileModule(code, {
-        fileName,
-        transformers,
-        compilerOptions: config.options,
-        reportDiagnostics: true,
-      });
-
-      const diagnosticList = result.diagnostics
-        ? filterDiagnostics(result.diagnostics, ignoreDiagnostics)
-        : [];
-
-      reportTSError(diagnosticList, config.options.noEmitOnError);
-
-      return { code: result.outputText, map: result.sourceMapText as string };
-    };
+  function getBuild(
+    configFileName = '',
+    skipTypeCheck?: boolean
+  ): GetOutputFunction {
+    let getOutput: GetOutputFunction;
+    let outFiles = new Map<string, SourceOutput>();
+    const cache = configFileToBuildMap.get(configFileName);
+
+    if (cache) {
+      getOutput = cache.getOutput;
+      outFiles = cache.outFiles;
```

Comment:
```suggestion
      return cache.getOutput;
```

It doesn't appear that `outFiles` is actually being used after this. The `else` is getting pretty large, so it would be easier for me to read if we could return sooner.

Reply:
@cb1kenobi It is used several times below. Take a look with whitespace removed https://github.com/vercel/vercel/pull/8987/files?w=1

Reply:
I agree with @cb1kenobi here. I think that the `else` is unnecessary and adds makes the resulting code (not just the diff) a bit harder to read.

Reply:
@styfle I opened this file up in VSCode so I could collapse the code and it doesn't appear that `outFiles` is being referenced in a meaningful way. I guess I don't understand where it's being used.

Reply:
@cb1kenobi @EndangeredMassa You both are correct. Since there is a closure, there is no need to cache the `outFiles`. I just push [a2faf98](https://github.com/vercel/vercel/pull/8987/commits/a2faf9865e425047ec4f1182536867a8e2650d56) with your suggestion 👍

Reply:
Change looks good! Now we just need to make the tests happy.

Reply:
@cb1kenobi The failing tests seem unrelated to my PR since I only changed `@vercel/node` and those are passing